rust/lance-encoding/src/decoder.rs

```diff
@@ -467,14 +466,14 @@ impl<'a> DecoderMiddlewareChainCursor<'a> {
 }
 
 pub struct ColumnInfoIter<'a> {
-    column_infos: &'a [ColumnInfo],
+    column_infos: Vec<Arc<ColumnInfo>>,
```

Comment:
I need the decoder to make a copy of the column metadata because it unwraps it (peels off the zone info) during decoding.